tests/ttnn/unit_tests/operations/test_moreh_norm.py

```diff
@@ -0,0 +1,328 @@
+    "input_shape",
+    [
+        [10, TILE_HEIGHT, TILE_WIDTH],
+    ],
+)
+@pytest.mark.parametrize("compute_kernel_options", compute_kernel_options, ids=compute_kernel_ids)
+def test_moreh_norm_compute_kernel_options(input_shape, p, dim_rtol_atol, compute_kernel_options, device):
+    torch.manual_seed(2024)
+
+    dim, rtol, atol = dim_rtol_atol
+
+    run_moreh_norm(input_shape, p, dim, rtol, atol, device, compute_kernel_options=compute_kernel_options)
+
+
+# @pytest.mark.parametrize("p", [2.0], ids=["p=2.0"])
+# @pytest.mark.parametrize(
+#     "dim_rtol_atol",
+#     [
+#         [0, 0.1, 0.1],
+#         [1, 0.1, 0.1],
+#         [2, 0.1, 0.1],
+#     ],
+# )
+# @pytest.mark.parametrize(
+#     "input_shape",
+#     [
+#         [10, TILE_HEIGHT, TILE_WIDTH],
+#     ],
+# )
+# def test_moreh_norm_callback(input_shape, p, dim_rtol_atol, device, use_program_cache):
+#     torch.manual_seed(2024)
+
+#     dim, rtol, atol = dim_rtol_atol
+
+#     for _ in range(2):
+#         run_moreh_norm(input_shape, p, dim, rtol, atol, device)
+
+
+# @pytest.mark.parametrize("p", [2.0], ids=["p=2.0"])
+# @pytest.mark.parametrize(
+#     "dim_rtol_atol",
+#     [
+#         [[], 0.2, 0.2],
+#         [None, 0.2, 0.2],
+#         [0, 0.1, 0.1],
+#         [1, 0.1, 0.1],
+#         [2, 0.1, 0.1],
+#         [3, 0.1, 0.1],
+#         [[0, 1], 0.1, 0.1],
+#         [[0, 1, 2], 0.15, 0.15],
+#         [[0, 1, 2, 3], 0.2, 0.2],
+#         [[0, 1, 3], 0.15, 0.15],
+#         [[0, 2, 3], 0.15, 0.15],
+#         [[1, 2], 0.1, 0.1],
+#         [[1, 2, 3], 0.15, 0.15],
+#         [[1, 3], 0.1, 0.1],
+#         [[2, 3], 0.1, 0.1],
+#     ],
+#     ids=[
+#         "global_norm(dim=[])",
+#         "global_norm(dim=None)",
+#         "N",
+#         "C",
+#         "H",
+#         "W",
+#         "NC",
+#         "NCH",
+#         "NCHW",
+#         "NCW",
+#         "NHW",
+#         "CH",
+#         "CHW",
+#         "CW",
+#         "HW",
+#     ],
+# )
+# @pytest.mark.parametrize(
+#     "input_shape",
+#     [
+#         [TILE_HEIGHT, TILE_WIDTH],
+#         [2, 2, 2 * TILE_HEIGHT + 13, 2 * TILE_WIDTH + 13],
+#     ],
+# )
+# @pytest.mark.parametrize("keepdim", [True, False], ids=["keepdim-true", "keepdim-flase"])
+# def test_moreh_norm_backward(input_shape, p, dim_rtol_atol, keepdim, device):
+#     torch.manual_seed(2024)
+
+#     dim, rtol, atol = dim_rtol_atol
+
+#     run_moreh_norm_backward(input_shape, p, dim, rtol, atol, device, keepdim=keepdim)
+
+
+# @pytest.mark.parametrize("p", [2.0], ids=["p=2.0"])
+# @pytest.mark.parametrize(
+#     "dim_rtol_atol",
+#     [
+#         [[], 0.2, 0.2],
+#     ],
+# )
+# @pytest.mark.parametrize(
+#     "input_shape",
+#     [[2, 2], [32, 2], [2, 32], [32, 32]],
+# )
+# @pytest.mark.parametrize("compute_kernel_options", compute_kernel_options, ids=compute_kernel_ids)
+# def test_moreh_norm_backward_compute_kernel_options(input_shape, p, dim_rtol_atol, compute_kernel_options, device):
+#     torch.manual_seed(2024)
```

Comment:
The backward code is still commented out.
Is this commented out because the backward work is not yet done?

Reply:
Yes, backward is on going.

Reply:
Do you think going backward in this PR is good, or would it be better in another PR?